assessment_factory/public/js/src/assessment_factory.js

```diff
@@ -1,22 +1,156 @@
 /* Javascript for AssessmentFactoryXBlock. */
-function AssessmentFactoryXBlock(runtime, element) {
+function AssessmentFactoryBlock(runtime, element, ctx) {
 
-    function updateCount(result) {
-        $('.count', element).text(result.count);
-    }
+    $(document).on('click', '.reset', function(){
+        $.ajax({
+            type: 'POST',
+            url: runtime.handlerUrl(element, 'reset'),
+            data: '{}',
+            success: function(data){
+                $(".item").remove();  
+                $(".category").remove();
+
+                renderElements(data.ctx);
+                initDraggable();
+                initDroppable();   
+                if($(".check").length){
+                    $(".check").text("Check (" + data.ctx.retry_count + " left)");
+                }
+                else{
+                    $(".other-container").append('<i class="check">Check (3 left)</i>');
+                    $(".other-container").append('<i class="submit">Submit</i>');
+                }
+                
+            }
+        });
+    });
+
+    $(document).on('click', '.check', function(){
+        $.ajax({
+            type: 'POST',
+            url: runtime.handlerUrl(element, 'check_problem'),
+            data: '{}',
+            success: function(data){
+                for(var key in data.incorrect_items){
+                    $("#"+data.incorrect_items[key]).addClass("incorrect-item");
+                }
```

Comment:
Razmisli o koristenju Underscora za ovo. On ima each koji je malo citljiviji, a sto se tice ovog selektora, mozes napisati pomocnu funkciju koja ce na temelju parametra vratiti string selector #{vrijednost}.

Imas to i u petlji ispod. Dakle isti komentar.